Sources/FeaturevisorSDK/Instance.swift

```diff
@@ -81,6 +99,46 @@ public struct Evaluation {
         self.variableValue = variableValue
         self.variableSchema = variableSchema
     }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(featureKey, forKey: .featureKey)
+        try container.encode(reason.rawValue, forKey: .reason)
+        try container.encode(bucketValue, forKey: .bucketValue)
+        try container.encode(ruleKey, forKey: .ruleKey)
+        if let error = error {
+            try container.encode(error.localizedDescription, forKey: .error)
+        }
+        try container.encode(enabled, forKey: .enabled)
+        try container.encode(traffic, forKey: .traffic)
+        try container.encode(sticky, forKey: .sticky)
+        try container.encode(initial, forKey: .initial)
+        try container.encode(variation, forKey: .variation)
+        try container.encode(variationValue, forKey: .variationValue)
+        try container.encode(variableKey, forKey: .variableKey)
+        try container.encode(variableValue, forKey: .variableValue)
+        try container.encode(variableSchema, forKey: .variableSchema)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        featureKey = try container.decode(FeatureKey.self, forKey: .featureKey)
```

Comment:
There are optional fields, so just `try` will throw an error if the field is missing. For optional ones, you should use `try?`